src/main/scala/ch/epfl/bluebrain/nexus/kg/resources/Rejection.scala

```diff
@@ -101,4 +110,11 @@ object Rejection {
     * @param message a human readable description of the cause
     */
   final case class DownstreamServiceError(override val message: String) extends Rejection(message)
+
+  /**
+    * Signals an unexpected rejection
+    *
+    * @param message a human readable description of the cause
+    */
+  final case class Unexpected(override val message: String) extends Rejection(message)
```

Comment:
maybe we should request a `F[_] : MonadError` and raise the unexpected failures with `F.raiseError` instead. It doesn't seem to be a user generated issue.

Thoughts?

Reply:
I didn't want to mix `MonadError` in some places with `EitherT` or `OptionT` in others. I think having a common one makes things cleaner. There is nowhere on the definition of a `Rejection` an assumption of user input errors only (DownsrteamRejection as another example of non user input error).

Reply:
I think longer term would be better to discriminate between system failures and user rejections.
We still have the case where we don't produce an EitherT value; for example in sourcing we produce a failed future equivalent to `F.raiseError`:

https://github.com/BlueBrain/nexus-sourcing/blob/master/modules/akka/src/main/scala/ch/epfl/bluebrain/nexus/sourcing/akka/ShardingAggregate.scala#L46